diff --git a/tools/packaging/test262.py b/tools/packaging/test262.py
index 921360a05e..a19c8a14e6 100755
--- a/tools/packaging/test262.py
+++ b/tools/packaging/test262.py
@@ -168,6 +168,8 @@ class TestResult(object):
     if len(err) > 0:
        target.write("--- errors ---  \n %s" % err)
 
+    target.write("\n--- exit code: %d ---\n" % self.exit_code)
+
   # This is a way to make the output from the "whitespace" tests into valid XML
   def SafeFormat(self, msg):
     try:
@@ -469,8 +471,8 @@ class TestSuite(object):
           if self.ShouldRun(rel_path, tests):
             basename = path.basename(full_path)[:-3]
             name = rel_path.split(path.sep)[:-1] + [basename]
-            if EXCLUDE_LIST.count(basename) >= 1:
-              print 'Excluded: ' + basename
+            if rel_path in EXCLUDE_LIST:
+              print 'Excluded: ' + rel_path
             else:
               if not self.non_strict_only:
                 strict_case = TestCase(self, name, full_path, True)
