diff --git a/apps/system/init/init.c b/apps/system/init/init.c
index 3aaa1860..815a9c11 100644
--- a/apps/system/init/init.c
+++ b/apps/system/init/init.c
@@ -139,6 +139,10 @@ int preapp_start(int argc, char *argv[])
 	}
 #endif
 
+#ifdef CONFIG_FS_ROMFS
+	mount("/dev/smart4rom9", "/rom", "romfs", 0, NULL);
+#endif
+
 #if defined(CONFIG_LIB_USRWORK) || defined(CONFIG_TASH)
 error_out:
 	return pid;
diff --git a/build/configs/artik053/artik053_download.sh b/build/configs/artik053/artik053_download.sh
index 0fa4b9a3..a3178545 100755
--- a/build/configs/artik053/artik053_download.sh
+++ b/build/configs/artik053/artik053_download.sh
@@ -111,6 +111,7 @@ main()
 			flash_write sssfw ../../bin/sssfw.bin; 		\
 			flash_write wlanfw ../../bin/wlanfw.bin;	\
 			flash_write os ../../../../output/bin/tinyara_head.bin;	\
+			flash_write rom ../../../../output/bin/rom.img;	\
 			exit' || finish_download 1
 			popd
 
diff --git a/build/configs/artik053/tools/openocd/partition_map.cfg b/build/configs/artik053/tools/openocd/partition_map.cfg
index c3abdeff..8dca6666 100644
--- a/build/configs/artik053/tools/openocd/partition_map.cfg
+++ b/build/configs/artik053/tools/openocd/partition_map.cfg
@@ -18,7 +18,8 @@ set partition_list {
 	os	{ "OS"			0x040C8000	0x00258000	0 }
 	factory	{ "Factory Reset"	0x04320000	0x00180000	0 }
 	ota	{ "OTA download"	0x044A0000	0x00180000	0 }
-	user	{ "USER R/W"		0x04620000	0x0015E000	0 }
+	user	{ "USER R/W"		0x04620000	0x000FA000	0 }
+	rom	{ "ROM FS"		0x0471A000	0x00064000	0 }
 	nvram	{ "WiFi NVRAM"		0x0477E000	0x00002000	1 }
 	sssrw	{ "SSS R/W Key"		0x04780000	0x00080000	1 }
 }
